h/sentry/subscribers.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import unicode_literals
+
+import traceback
+
+from pyramid.events import subscriber
+from pyramid_retry import IBeforeRetry
+import sentry_sdk
+
+
+@subscriber(IBeforeRetry)
+def add_retryable_error_to_sentry_context(event):
+    """
+    Add information about a retryable error to the Sentry context.
+
+    If a request raises a retryable error then, if we're not already on this
+    request's last retry attempt, pyramid_retry calls this IBeforeRetry
+    subscriber function before it retries the request.
+
+    This function adds some information to the Sentry context about the
+    retryable error that was raised by the failed attempt at the request.
+
+    If a future retry of this request raises a non-retryable error, or if we
+    use up the maximum number of retry attempts, so that we ultimately end up
+    sending an error response back to the client, then the information that
+    this function adds to the Sentry context will be included in the event that
+    is ultimately reported to Sentry.
+
+    This makes debugging errors easier because you can see information about
```

Comment:
This would only be useful if the final exception was different than the exceptions before it. Is there a particular issue that lead you to add this? Personally I've found the exceptions to generally be all the same in which case I'm not sure this would really be useful.

Reply:
Thinking about the `ConcurrentUpdateError` that we're seeing on prod here, and just trying to add as much debugging info to those Sentry reports as possible, 'cos I still don't get why that exception is happening. This info will enable us to see when a request that got reported to Sentry had previous failed retries (which I don't believe we can see in Sentry) and some info about those previous fails. It'll also let us see if `ConcurrentUpdateError` is getting erroneously reported on the first and second failed attempts not just on the final attempt (which it shouldn't be, on this branch, but good to be able to see for sure)